sct.py

```diff
@@ -484,6 +487,79 @@ def list_repos(dist_type, dist_version):
     click.echo(tbl.get_string(title="Scylla Repos"))
 
 
+@cli.command('get-scylla-base-versions', help='Get Scylla base versions of upgrade')
+@click.option('-s', '--scylla-version', type=str,
+              help='Scylla version, eg: 4.5, 2021.1')
+@click.option('-r', '--scylla-repo', type=str,
+              help='Scylla repo')
+@click.option('-d', '--linux-distro', type=str,
+              default='centos', help='Linux Distribution type')
+@click.option('-o', '--only-print-versions', type=bool, default=False, required=False, help='')
+def get_scylla_base_versions(scylla_version, scylla_repo, linux_distro, only_print_versions):  # pylint: disable=too-many-locals
+    """
+    Upgrade test try to upgrade from multiple supported base versions, this command is used to
+    get the base versions according to the scylla repo and distro type, then we don't need to hardcode
+    the base version for each branch.
+    """
+    add_file_logger()
+
+    dist_type = linux_distro.split('-')[0]
+    dist_version = linux_distro.split('-')[-1]
+
+    if scylla_version is None:
+        assert 'unstable/' in scylla_repo
+        product, version = scylla_repo.split('unstable/')[1].split('/')[0:2]
+        scylla_version = version.replace('branch-', '').replace('enterprise-', '')
+    else:
+        product = 'scylla-enterprise' if is_enterprise(scylla_version) else 'scylla'
+
+    if not dist_type == 'centos' and dist_version is None:
+        click.secho("when passing --dist-type=debian/ubuntu need to pass --dist-version as well", fg='red')
+        sys.exit(1)
+
+    # We can't detect the support versions for this distro, which shares the repo with others, eg: centos8
+    # so we need to assign the start support versions for it.
+    oos_start_support_version = None
+    ent_start_support_version = None
+    if (dist_type not in ['debian', 'ubuntu'] and linux_distro != 'centos') and (
+            oos_start_support_version is None or ent_start_support_version is None):
+        versions_dict = start_support_versions.get(linux_distro, None)
+        if versions_dict is None:
+            click.secho(
+                f"We can't detect the support versions for this distro, which shares the repo with {dist_type}.")
+            sys.exit(1)
+        oos_start_support_version = versions_dict['scylla']
+        ent_start_support_version = versions_dict['enterprise']
+
```

Comment:
Please place this code it into the function and put it into `utils/get_supported_scylla_base_versions.py`